sklearn/linear_model/coordinate_descent.py

```diff
@@ -99,9 +99,12 @@ def _alpha_grid(X, y, Xy=None, l1_ratio=1.0, fit_intercept=True,
             Xy /= X_std[:, np.newaxis]
     alpha_max = (np.sqrt(np.sum(Xy ** 2, axis=1)).max() /
                  (n_samples * l1_ratio))
-    alphas = np.logspace(np.log10(alpha_max * eps), np.log10(alpha_max),
-                         num=n_alphas)[::-1]
-    return alphas
+
+    if alpha_max == 0:
+        return np.zeros(n_alphas)
```

Comment:
I meant checking if `alpha_max` is 0 at the end instead of checking `if np.unique(y).size == 1` ( which is one of the possible ways in which `alpha_max` can be 0 ). :)